### ISSUE TITLE
radiologist-inference CLI: registry selectors + serve + health split + bugfixes

### ISSUE THREAD

**Issue by CedrickArmel** [CedrickArmel/radiologist#125]
## ✨ radiologist-inference CLI: registry selectors + serve + health split + bugfixes

**Depends on:** #1 (skeleton), #2 (RegistrySelector + resolve_selector). **Blocks:** none.

### Problem

The inference CLI can only load models from a local `--model` path; there is no way to pull a model
from the W&B registry, no `serve` command to launch the HTTP API, and `/healthz` conflates liveness
with readiness (it currently 503s when no predictor is loaded). Two adjacent bugs also live in this
package: a Score-CAM docstring that misdescribes its own inputs, and a wandb-missing error message
that names the wrong extra for inference users.

### Goal

`predict`/`explain`/`uncertainty` accept registry selectors and dispatch to the registry or a local
file; a new `serve` command launches FastAPI via uvicorn; `/healthz` is pure liveness and a new
`/readyz` owns readiness; and the two bugs are fixed.

### User story

As an **operator**, I want to run inference and serve the API directly against models in the W&B
registry, and I want liveness and readiness probes to mean what their names say.

### Context a reader needs (inline)

The skeleton (#1) already added: the selector flags to `predict`/`explain`/`uncertainty`, the `serve`
command stub, `BasePredictor.from_selector(...)` stub, the `_uvicorn` sentinel in
`radiologist.inference.optional`, and a `/readyz` route stub in `app.py`.

**`from radiologist.registry import RegistrySelector, resolve_selector`** — `RegistrySelector` is a
frozen dataclass with `path, run_id, groups, tags, metric, version, include_sweeps` and a method
`is_registry_backed() -> bool` (True when any registry field is set). `resolve_selector(selector,
registry) -> ArtifactRef` applies the `run_id → tags → path` cascade and raises `ValueError` when both
`run_id` and `tags` are given.

**Predictor classes** (in `radiologist.inference`): `Classifier`, `Explainer`, `MCDropoutPredictor`,
all subclassing `BasePredictor`. Loading surface:

```python
BasePredictor.from_path(det_path: str, mcd_path: Optional[str] = None) -> BasePredictor
BasePredictor.from_registry(artifact_path: str, local_dir: str,
                            registry: Optional[ModelRegistry] = None) -> BasePredictor  # UNCHANGED
```

`from_selector(selector, local_dir, registry=None)` was stubbed in #1 and is implemented here:
`resolve_selector(selector, registry or WandbRegistry())` → `registry.download(ref, local_dir)` →
`cls.from_path(det_path=<downloaded path>)`.

**Current CLI command bodies** (for the local `--model` path that must keep working):
- `predict`: `Classifier.from_path(det_path=model)` → `.predict(image=image_path)` → echo predicted class + probabilities.
- `explain`: `Explainer.from_path(det_path=model)` → `.explain(image=image_path)` → echo predicted class, save/echo saliency.
- `uncertainty`: `MCDropoutPredictor.from_path(det_path=model, mcd_path=mcd_model)` → `.predict_with_uncertainty(image=image_path, n_passes=n_passes)`.

**Current `app.py` health route** (to be split):

```python
@app.get("/healthz")
def healthz() -> Dict[str, str]:
    _get_predictor()   # raises HTTPException(503) when no predictor loaded
    return {"status": "ok"}
```

`_get_predictor()` raises `HTTPException(status_code=503, detail="No model loaded. ...")` when the
injected predictor is `None`, else returns it. `create_app(predictor=None)` builds the app.

### Implementation

**1. Selector dispatch in `predict`/`explain`/`uncertainty`.** Build a `RegistrySelector` from the
`--run-id`/`--tags`/`--groups`/`--metric` flags (with `path=model or ""`). If
`selector.is_registry_backed()`, load via `<PredictorClass>.from_selector(selector, local_dir=local_dir)`;
else if `--model` is set, load via `from_path(model)`; else raise a clear error (`typer` exits
non-zero) stating that one of `--model` or a registry selector is required. `--tags` and `--groups`
are repeatable options (`List[str]`), not comma-separated.

**2. `uncertainty` registry path.** When registry-backed, resolve the deterministic model via the
given `--run-id` and the MC-Dropout model via the `{run_id}-mcd` convention (this mirrors
`WandbRegistry.promote()`, which internally resolves `mcd_ref = resolve(path, run_id=f"{run_id}-mcd",
version="best")`). Pull both, then `MCDropoutPredictor.from_path(det_path=<det>, mcd_path=<mcd>)`.
(Local path stays `--model` + `--mcd-model` as today.)

**3. `serve` command.** Build a predictor: `from_selector(...)` when the selector is registry-backed,
else `from_path(model)` when `--model` is set, else `None` (lazy routes). Then
`uvicorn.run(create_app(predictor), host=host, port=port)`, reaching uvicorn through the `_uvicorn`
sentinel. Raise `RuntimeError` naming the `serve` extra when `_uvicorn is None`.

**4. `/healthz` vs `/readyz` split** in `app.py`'s `_build_app`:
- `/healthz` becomes pure liveness — always returns `{"status": "ok"}` with 200, never calls
  `_get_predictor()`.
- `/readyz` takes over the readiness behavior — calls `_get_predictor()` so it returns 503 when no
  predictor is loaded and 200 (`{"status": "ready"}`) once one is.
- **This changes existing `/healthz` test expectations** (previously `/healthz` 503'd without a
  predictor). Update those tests to assert the new split: `/healthz` always 200, `/readyz` carries the
  503-when-empty behavior.

**5. Bugfix (a) — Score-CAM docstrings (docs only, no behavior change).**
- `radiologist-inference/src/radiologist/inference/cam.py`, `score_cam(feature_maps, logits)`: the
  docstring currently says it computes a saliency map "from feature maps and logits", but the
  implementation ignores `logits` entirely — it uses global-average-pooling of each channel as the
  channel weight (`weights = feature_maps.mean(axis=(1, 2))`). Correct the docstring to document
  `logits` as the unused GAP-approximation fallback path (used when no ONNX session is available for
  masked passes), so it stops implying `logits` affects the result.
- `radiologist-inference/tests/test_score_cam.py` module docstring currently claims
  `Explainer.explain()` "exercises the same saliency computation". It does not — `Explainer.explain()`
  uses the different `score_cam_with_session()` (which runs real masked ONNX passes), not
  `score_cam()`. Correct the test docstring to stop claiming parity. No test assertions change.

**6. Bugfix (b) — corrected wandb-missing extra message.** When `wandb` is absent, the registry layer
raises `RuntimeError` whose message names `radiologist-registry[wandb]` (from
`radiologist.registry.optional._WANDB_MISSING_MSG`). For inference users that extra name is wrong —
this package's own `pyproject.toml` declares the extra as `registry = ["wandb>=0.27.0"]`. In
`from_registry` and `from_selector`, catch that `RuntimeError` and re-raise with a corrected message
naming `radiologist-inference[registry]`. **Do not** modify the registry package's own message — it
is correct for registry's own CLI users.

### Acceptance criteria

Drive the real Typer `app` via `CliRunner`; drive the FastAPI app via a real test client from
`create_app(...)`. Mock only true boundaries: the W&B SDK at the `_wandb` sentinel, and `uvicorn` at
the `_uvicorn` sentinel. Never mock `Classifier`/`Explainer`/`MCDropoutPredictor`/`WandbRegistry` or
any owned class.

- [ ] `predict IMG --model PATH` loads the local file and prints the predicted class (unchanged behavior).
- [ ] `predict IMG --run-id R` resolves the model for run R from the registry, loads it, and prints the predicted class.
- [ ] `predict IMG --tags a --tags b` passes both tags as a list to resolution (repeatable option, not comma-split).
- [ ] `predict IMG` with neither `--model` nor any selector flag exits non-zero with a clear message naming both options.
- [ ] `predict IMG --run-id R --tags a` exits non-zero (both run_id and tags rejected by `resolve_selector`).
- [ ] `uncertainty IMG --run-id R` resolves the deterministic model for run R and the MC-Dropout model for `R-mcd`, and prints the predictive entropy.
- [ ] `serve --model PATH` builds the app with the loaded predictor and invokes the server runner on the given host/port (uvicorn boundary mocked).
- [ ] `serve` with `_uvicorn` absent raises a `RuntimeError` naming the `serve` extra.
- [ ] `GET /healthz` returns 200 whether or not a predictor is loaded.
- [ ] `GET /readyz` returns 503 when no predictor is loaded and 200 once one is.
- [ ] When `wandb` is absent, `predict IMG --run-id R` fails with a message naming `radiologist-inference[registry]` (not `radiologist-registry[wandb]`).
- [ ] `cam.py`'s `score_cam` docstring no longer claims `logits` affects the result; `test_score_cam.py`'s module docstring no longer claims parity with `Explainer.explain()`.
- [ ] mypy clean; pytest green.
